astropy/coordinates/representation.py

```diff
@@ -421,39 +421,56 @@ class CartesianRepresentation(BaseRepresentation):
 
     Parameters
     ----------
-    x, y, z : `~astropy.units.Quantity`
-        The x, y, and z coordinates of the point(s). If ``x``, ``y``, and
-        ``z`` have different shapes, they should be broadcastable.
-
+    x, y, z : `~astropy.units.Quantity` or array
+        The x, y, and z coordinates of the point(s). If ``x``, ``y``, and ``z``
+        have different shapes, they should be broadcastable. If not quantity,
+        ``unit`` should be set.  If only ``x`` is given, it is assumed that it
+        contains an array with the 3 coordinates are stored along ``axis``.
+    unit : `~astropy.unit.Unit` or str
+        If given, the coordinates will be converted to this unit (or taken to
+        be in this unit if not given.
+    axis : int, optional
+        The axis along which the coordinates are stored for the case an array
+        of vectors is passed in (default: 0).
     copy : bool, optional
-        If True arrays will be copied rather than referenced.
+        If True (default), arrays will be copied rather than referenced.
     """
 
     attr_classes = OrderedDict([('x', u.Quantity),
                                 ('y', u.Quantity),
                                 ('z', u.Quantity)])
 
-    def __init__(self, x, y=None, z=None, copy=True):
+    def __init__(self, x, y=None, z=None, unit=None, axis=0, copy=True):
```

Comment:
Here (and in the `get_xyz`), I think it might be better to call it something like `xyz_axis`.  With just "axis" it's not really clear at a glance what that means.